src/spark_rapids_ml/knn.py

```diff
@@ -422,32 +482,17 @@ async def do_allGather() -> List[str]:
 
         return _cuml_fit
 
-
-class NearestNeighborsModel(
-    NearestNeighborsClass, _CumlModel, _NearestNeighborsCumlParams
-):
-    def __init__(
-        self,
-        query_index: List[int],
-        distances: List[List[float]],
-        indices: List[List[int]],
-    ):
-        super().__init__(
-            query_index=query_index,
-            distances=distances,
-            indices=indices,
+    def _transform(self, dataset: DataFrame) -> DataFrame:
+        raise NotImplementedError(
+            "'NearestNeighborsModel does not provide a transform function. Use 'kneighbors' instead."
         )
 
-        cumlParams = NearestNeighbors._get_cuml_params_default()
-        self.set_params(**cumlParams)
-
-    def _out_schema(self, input_schema: StructType) -> Union[StructType, str]:  # type: ignore
-        pass
-
-    def _get_cuml_transform_func(  # type: ignore
+    def _get_cuml_transform_func(
         self, dataset: DataFrame
     ) -> Tuple[
         Callable[..., CumlT],
         Callable[[CumlT, Union[cudf.DataFrame, np.ndarray]], pd.DataFrame],
     ]:
-        pass
+        raise NotImplementedError(
+            "'_CumlModel._get_cuml_transform_func' method is not implemented. Use 'kneighbors' instead."
```

Comment:
Is this error message for the end user?

Reply:
Yeah. Currently, the Model class of KNN does not support the transform function.